internal/server/httpserver/server.go

```diff
@@ -0,0 +1,17 @@
+package httpserver
+
+import (
+	"github.com/Osselnet/metrics-collector/internal/server/handlers"
+	"net/http"
+)
+
+type Config struct {
+	Addr string
+}
+
+func New(h *handlers.Handler, cfg Config) {
```

Comment:
Тут конфиг для пакета лишний, проще явно передать один параметр
```suggestion
func New(h *handlers.Handler, addr string) {
```
А так запускать сервер приветствуется из main и тогда этот пакт не нужен. 
Идет обращение к пакету http без какой-то либо логики